### PR TITLE
docs(architecture): create Session actor doc

### PR DESCRIPTION
**Pull request by lrubiorod**
Create Session actor documentation with the currently functionality

### CODE REVIEW

**Comment by aesedepece** [comment]
@jrmoreau r? for copies :pray:

File: docs/architecture/session.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## API
+
+### Outgoing messages: Sessions Manager -> Others
+
+These are the messages sent by the connections manager:
+
+| Message               | Destination             | Input type                                        | Output type                       | Description                           |
+|-----------------------|-------------------------|---------------------------------------------------|-----------------------------------|---------------------------------------|
+| Register              | SessionsManager         | `SocketAddr, Addr<Session>, SessionType`          | `SessionsResult<()>`              | Request to register a new session     |
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
| `Register`              | `SessionsManager`         | `SocketAddr, Addr<Session>, SessionType`          | `SessionsResult<()>`              | Request to register a new session     |
```

File: docs/architecture/session.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+| Unregister            | SessionsManager         | `SocketAddr, SessionType`                         | `SessionsResult<()>`              | Request to unregister a session       |
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
| `Unregister`            | `SessionsManager`         | `SocketAddr, SessionType`                         | `SessionsResult<()>`              | Request to unregister a session       |
```

File: docs/architecture/session.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+#### Register
+
+This message is sent to the [`Sessions Manager`][sessions_manager] actor when the session 
+actor is started to register this session in the hash map of
```
Comment (aesedepece) [inline docs/architecture/session.md]: This actor shouldn't care about the specific data structure that `SessionsManager` is using internally for storing references to sessions, and so shouldn't its documentation.

File: docs/architecture/session.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+| Unregister            | SessionsManager         | `SocketAddr, SessionType`                         | `SessionsResult<()>`              | Request to unregister a session       |
+
+
+#### Register
+
+This message is sent to the [`Sessions Manager`][sessions_manager] actor when the session 
+actor is started to register this session in the hash map of 
+[`Sessions Manager`][sessions_manager].
+
+The return value is a Result to check if the operation has been done.
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
The returned value is a `Result` for easy verification of the success of the operation.
```

File: docs/architecture/session.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+| Message               | Destination             | Input type                                        | Output type                       | Description                           |
+|-----------------------|-------------------------|---------------------------------------------------|-----------------------------------|---------------------------------------|
+| Register              | SessionsManager         | `SocketAddr, Addr<Session>, SessionType`          | `SessionsResult<()>`              | Request to register a new session     |
+| Unregister            | SessionsManager         | `SocketAddr, SessionType`                         | `SessionsResult<()>`              | Request to unregister a session       |
+
+
+#### Register
+
+This message is sent to the [`Sessions Manager`][sessions_manager] actor when the session
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
This message is sent to the [`SessionsManager`][sessions_manager] actor when the session 
```

File: docs/architecture/session.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+#### Register
+
+This message is sent to the [`Sessions Manager`][sessions_manager] actor when the session 
+actor is started to register this session in the hash map of 
+[`Sessions Manager`][sessions_manager].
+
+The return value is a Result to check if the operation has been done.
+
+For further information, see [`Sessions Manager`][sessions_manager].
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
For further information, see [`SessionsManager`][sessions_manager].
```

File: docs/architecture/session.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+#### Unregister
+ 
+This message is sent to the [`Sessions Manager`][sessions_manager] actor when the session
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
This message is sent to the [`SessionsManager`][sessions_manager] actor when the session
```

File: docs/architecture/session.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+actor is stopped to unregister this session in the hash map of the
```
Comment (aesedepece) [inline docs/architecture/session.md]: This actor shouldn't care about the specific data structure that `SessionsManager` is using internally for storing references to sessions, and so shouldn't its documentation.

File: docs/architecture/session.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+For further information, see [`Sessions Manager`][sessions_manager].
+
+#### Unregister
+ 
+This message is sent to the [`Sessions Manager`][sessions_manager] actor when the session
+actor is stopped to unregister this session in the hash map of the 
+[`Sessions Manager`][sessions_manager].
+
+The return value is a Result to check if the operation has been done.
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
The returned value is a `Result` for easy verification of the success of the operation.
```

File: docs/architecture/session.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+For further information, see [`Sessions Manager`][sessions_manager].
+
+## Further information
+The full source code of the `Session` can be found at [`session.rs`][session].
```
Comment (aesedepece) [inline docs/architecture/session.md]:
```suggestion
The full source code of the `Session` actor can be found at [`session.rs`][session].
```

File: docs/architecture/session.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
+# Session
+
+__Session__ is the actor that encapsulates the logic of the sessions 
+associated to each TCP connections.
```
Comment (aesedepece) [inline docs/architecture/session.md]: - This actor actually encapsulates the entire business logic of the Witnet network protocol.
- This actor shouldn't care about the specific transport used by the `ConnectionsManager`, ans so shouldn't its documentation.

File: docs/architecture/session.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+
+
+## Actor creation and registration
+
+The creation of the session actor and its registration into the system registry are
```
Comment (jrmoreau) [inline docs/architecture/session.md]: The creation of the session actor and its addition to the system registry are
performed by the `connection_manager` process:

File: docs/architecture/session.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    // Get peer address
+    let address = stream.peer_addr().unwrap();
+
+    // Split TCP stream into read and write parts
+    let (r, w) = stream.split();
+
+    // Add stream in session actor from the read part of the tcp stream
+    Session::add_stream(FramedRead::new(r, P2PCodec), ctx);
+
+    // Create the session actor and store in its state the write part of the tcp stream
```
Comment (jrmoreau) [inline docs/architecture/session.md]: // Create the session actor and store in its state the write part of the TCP stream

**Comment by lrubiorod** [comment]
Suggestions applied

**Pull request merged**